Clients/GUI/Shared/Header.razor

```diff
@@ -1,38 +1,90 @@
+        <a class="navbar-brand pl-5" href="/">
+            <img src="Picture/Logo.png" width="30" height="30" class="d-inline-block align-top" alt="">
+        </a>
+        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
+            <span class="navbar-toggler-icon"></span>
+        </button>
+        <div class="collapse navbar-collapse" id="navbarNav">
+            <ul class="navbar-nav mr-auto">
+                <li class="nav-item">
+                    <a class="nav-link" href="/">News</a>
+                </li>
+                <AuthorizeView>
+                    <Authorized>
+                        <li class="nav-item">
+                            <a class="nav-link" href="/instruments">Instruments</a>
+                        </li>
+                    </Authorized>
+                </AuthorizeView>
+            </ul>
+            <div class="navbar-nav ml-auto">
+                <AuthorizeView>
+                    <NotAuthorized Context="NotAuth">
+                        <button type="button" class="btn btn-outline-secondary mr-2" @onclick="@(() => modal.Show<GUI.Pages.SignIn>())">Sign In</button>
+
+                        <button type="button" class="btn btn-outline-secondary" @onclick="@(() => modal.Show<GUI.Pages.SignUpWizzard>())">Sign up</button>
+                    </NotAuthorized>
+                    <Authorized>
+                        @{
+                            LoadImageSrc();
+                        }
+                        <div class="btn-group dropleft">
+                            <a class="navbar-brand" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                                <img src="@imgSrc" width="30" height="30" class="d-inline-block align-top rounded-circle" alt="Avatar">
+                            </a>
+
+                            <div class="dropdown-menu">
+                                <a class="dropdown-item align-items-center" href="userinfo">
+                                    <a>User Info</a>
+                                    <img src="@imgSrc" width="30" height="30" class="align-items-center rounded-circle" alt="Avatar">
+                                </a>
+                                <div class="dropdown-divider"></div>
+                                <a class="dropdown-item" href="/" @onclick="SignOut">Sign out</a>
+                            </div>
+                        </div>
+                    </Authorized>
+                </AuthorizeView>
+            </div>
+        </div>
+    </nav>
 </div>
 
 @code {
+    private string defaultImgSrc = "Picture/Avatar.svg";
+    private string imgSrc = "Picture/Avatar.svg";
+
     private void SignOut()
     {
         Storage.ClearAsync();
+
         AuthenticationProvider.MarkSignedOut();
     }
-}
 
+    private async void LoadImageSrc()
```

Comment:
I suggest moving this logic to OnAfterRender, because now we can't get image on first page load because you get items from storage only after rendering the page. So you may create a variable for image and change it in onafterrender

Reply:
I might be wrong a bit in comment above
Another way is to notify the component that it has changed after loading img

Reply:
> I might be wrong a bit in comment above
> Another way is to notify the component that it has changed after loading img

Just add `StateHasChanged();` at the end of image loading method in component

Reply:
added